drivers/hmis/app/models/hmis/ce/match/match_applicability.rb

```diff
@@ -0,0 +1,99 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+# The `MatchApplicability` class determines whether a Coordinated Entry (CE) rule applies to a given entity
+# (e.g., Opportunity, Unit, Unit Group, Project, or Organization) based on project criteria.
+#
+# This is NOT determining client is eligibility itself; rather it determines which rules to use for eligibility/priority.
+#
+# This class evaluates applicability rules for entities by checking:
+# - Whether the rule's owner matches the entity or one of its parent entities.
+# - Whether the entity's project type matches the rule's allowed project types.
+# - Whether the entity's project funders match the rule's allowed funders.
+#
+# Note:
+# - This class does NOT determine client eligibility itself; it determines which rules
+#   should be used for eligibility/priority.
+# - It supports evaluation against various entities (eg Project, Unit, etc) to support viewing applicable
+#   rules for each entity type.
+# - Global and Data Source-level rules are not yet supported (e.g. "all PSH projects in a data source")
+#
+#
+#
+# @attr [Object] owner Entity that owns/manages this applicability rule
+# @attr [Array<String>] project_types List of valid project types
+# @attr [Array<String>] project_funders List of valid project funder IDs
+module Hmis::Ce::Match
+  MatchApplicability = Struct.new(:owner, :project_types, :project_funders, keyword_init: true) do
```

Comment:
This replaces the `OpportunityApplicability` class. The main difference is that it can resolve applicability for any entity type, not just opportunities.

So you can ask:
* Does this rule apply to this Project?
* Does this rule apply to this Unit Group?
* etc.

This allows us to support future interfaces where we can display applicable rules on the Project page, on the Unit Group page, etc. As opposed to just on the Unit page.